### PR TITLE
Merge two similar iterator functions in OapDataFile

### PR DESCRIPTION
**Pull request by carsonwang**
## What changes were proposed in this pull request?
The codes in two iterator function are similar. Simplify the code in a single internal function.
We may also change the base DataFile class to only have a single iterator function.

## How was this patch tested?
Existing tests.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala [hunk 1]
```diff
@@ -111,66 +111,52 @@ private[oap] case class OapDataFile(path: String, schema: StructType,
...
-
-          CompletionIterator[InternalRow, Iterator[InternalRow]](iterator,
-            fiberCacheGroup.zip(requiredIds).foreach {
-              case (fiberCache, id) => closeRowGroup(DataFiber(this, id, groupId), fiberCache)
-            }
-          )
-      }
-    CompletionIterator[InternalRow, Iterator[InternalRow]](iterator, close())
+  def iterator(
+      conf: Configuration,
```
Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala]: as #553 said, should we use conf of DataFile and remove this parameter ?

Comment (carsonwang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala]: Sounds good to use config defined in DataFile. I think we can do that in a separated PR as you also mentioned other improvements we can do for DataFile API.

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala [hunk 2]
```diff
@@ -111,66 +111,52 @@ private[oap] case class OapDataFile(path: String, schema: StructType,
...
-    val row = new BatchColumn()
-    val iterator =
-      (0 until meta.groupCount).iterator.flatMap { groupId =>
-        val fiberCacheGroup = requiredIds.map(id =>
-          FiberCacheManager.get(DataFiber(this, id, groupId), conf))
-
-        val columns = fiberCacheGroup.zip(requiredIds).map { case (fiberCache, id) =>
-          new ColumnValues(meta.rowCountInEachGroup, schema(id).dataType, fiberCache)
-        }
+  private def iterator(
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala]: maybe a different name?

Comment (carsonwang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala]: What do you suggest for the new name, any candidate?

Comment (linhong-intel) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala]: @carsonwang @adrian-wang 
Can you think about a name and merge this? Simple PR shouldn't be pending so long time.
My suggestion is to change to `buildIterator()`

Comment (adrian-wang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/io/OapDataFile.scala]: good to me

**Comment by carsonwang** [comment]
Thanks @linhong-intel @adrian-wang . I've resolved the conflicts and updated the method name.

**Pull request merged**